### DISCUSSION TITLE
kerberos doesn't work in FreeRDP3

### DISCUSSION THREAD

**Discussion by tuxzer92** [FreeRDP/FreeRDP#10216] [Q&A]
I have RED OS installed.
Version FreeRDP: 3.4.0
OS in the domain Active directory 2012.
Logged into the OS as a domain user.
Got TGT

I'm trying to connect to the RDS server via Kerberos, but for some reason it asks for a password:
`[DEBUG][com.winpr.utils] - [SamOpen]: Could not open SAM file!`
Why is SAM file being searched for if Kerberos is needed?
[error.txt](https://github.com/FreeRDP/FreeRDP/files/15433723/error.txt)

**Comment by hardening** [comment]
Right now the Kerberos logon doesn't use an existing ccache and always ask for credentials: mostly because by default you don't connect in restricted admin mode, and so we need to know the credentials to pass them to the server.

**Reply by grasduinen** [reply]
Hi , same issue here. We have RHEL clients that are part of on IDM, that trusts the AD. An AD user receives an TGT from the AD login on the RHEL client. We can request an service ticket for TERMSRV, but still have to enter the password when using FreeRDP 3.5.1.
The user is already authenticated, so Kerberos SSO should be possible if we could some how parse the ticket.
Any help on this would be appreciated 👍

**Comment by tuxzer92** [comment]
Shouldn't he be using a keytab file?

**Comment by hardening** [comment]
keytab or credentials allows to get a TGT. But when going forward in the RDP nego you also needs the user/domain/password so that you can do SSO on the target server (unless you connect with restricted admin that remove that SSO capability).

**Comment by tuxzer92** [comment]
I apologize, but why is Kerberos support implemented in freerdp?
Isn't it for SSO? 
Is there any way to prevent the user from entering their password every time they connect to RDS?

**Reply by hardening** [reply]
In theory if you connect with the restricted admin flag, then the credential aren't needed and you can connect with materials from the ccache. But that's not the actual behavior right now (and you loose SSO on the target server's session when doing restricted admin).
I'm currently working on remote credential guard that has the same use-case of connecting from an established ccache without credentials, perhaps I could push the changes so that you can connect with restricted admin flag just without credentials (that would imply disabling the NTLM SSPI in SPNego).

**Reply by grasduinen** [reply]
I would like to test this first remote credential guard functionality, or restricted admin mode for that sense. To my understanding, will the FreedRDP client for the initial authentication then fetch the ticket already present on the (RHEL) OS?  Thanks!

**Reply by steelman** [reply]
@hardening

FWIW, I've created a [patch](https://github.com/rdesktop/rdesktop/pull/403) for rdesktop… oh, you already know about it (-;

**Reply by hardening** [reply]
@steelman FYI https://github.com/FreeRDP/FreeRDP/pull/10549 has remote credential guard implemented
